keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -0,0 +1,1215 @@
+NetworkPolicy to their Namespaces which isolates the workloads.
+
+AdminNetworkPolicy `Pass` rules allows an admin to delegate security posture for
+certain traffic to the Namespace owners by overriding any lower priority Allow or 
+Deny rules. For example, allowing intra-tenant traffic can be delegated to tenant 
+admins explicitly by the cluster admin with the use of `Pass` rules.
+
+#### Priority
+
+The policy instances will be ordered based on the numeric priority assigned to the
+ANP. `Priority` is a 32 bit integer value, where a larger number corresponds to 
+a higher precedence. The lowest "regular" value (see below for 
+special cases) is "1", which corresponds to the lowest "importance". High values 
+have higher importance. For alpha, this API defines "1000" as the maximum value, 
+but this may be revisited as the proposal advances. For future-safety, clients may 
+assume that higher values will eventually be allowed, and simply treat it as an int32.
+Any positive priority will have higher precedence over the namespaced NetworkPolicy 
+instances in the cluster, unless the traffic matches on a `Pass` rule that allows 
+it to bypass any lower positive priority ANP rules.
+
+Additionally, the special priority "0" can be used in the priority field to indicate
+that the rules in that policy instance shall be created at a priority lower than the
+Namespaced NetworkPolicies. 
+
+The relative precedence of the rules within a single ANP object (all of which 
+share a priority) will be determined by the order in which the rule is written. 
+Thus, a rule that appears at the top of the ingress/egress rules would take the 
+highest precedence. The maximum number of rules, which will be calculated as the 
+total summation of the AdminNetworkPolicyIngressRules and AdminNetworkPolicyEgressRules
+in a single ANP instance, will be 100.
+
+Conflict resolution: Two policies are considered to be conflicting if they are assigned the
+same `priority` and apply to the same resources or a union of resources. In order to avoid such conflicts, 
+we propose to include tooling for ANP resources to help alert the admin to potentially ambiguous ANP
+priority scenarios, more details in [risks and mitigation](#risks-and-mitigation). However, ultimately
+it will be the job of the network policy implementation to decide how to handle overlapping priority 
+situations. 
+
+#### Rule Names
+
+In order to help future proof the ANP api, a built in mechanism to identify each
+allow/deny/pass rule is required. Such a mechanism will help administrators organize 
+and identify individual rules within an AdminNetworkPolicy resource.
+We propose to introduce a new string field, called `name`, in each `AdminNetworkPolicy`
+ingress/egress rule. Currently the `name` of a rule is optional and does not need to 
+be unique, due to the fact that there's no easy way to validate this in-tree. It 
+is suggested that implementations provide a mechanism to validate any name 
+duplications.
```

Comment:
I assume this section means that the "Rule.Name" field needs to be globally unique? That goes against the k8s API general guidelines. As Tim observes, the ANP name plus the rule name is guaranteed to be unique. Is that not sufficient?

Reply:
So yeah the rule name itself does not need to be globally unique, only unique within each ANP... We were under the assumption however that it would be impossible to verify that without an admission webhook, which is not something we can build in here.  

Is that assumption incorrect?

Reply:
Yes, that's correct. Only ObjectMeta.Name is guaranteed to be globally unique.